internal/pkg/manifest/validate.go

```diff
@@ -2042,3 +2045,8 @@ func contains(name string, names []string) bool {
 	}
 	return false
 }
+
+// validate is a no-op for SidecarImageConfig.
+func (s SidecarImageConfig) validate() error {
```

Comment:
This is the place where you should be testing what we are testing inside `UnmarshalYAML`